### ISSUE TITLE
Allow non-marginalized mixtures

### ISSUE THREAD

**Issue by larryshamalama** [pymc-devs/pymc#5717]
Currently, `pm.Mixture` only implements marginalized mixtures, i.e. index variables are not sampled. While this approach allows gradient-based sampling, PyMC default sampling mechanism, users may be interested in the latent variable that indexes components despite a generally slower sampling time. Most of the code under `rv_op` will still be needed and we would integrate the mixture functionality provided in [AePPL](https://github.com/aesara-devs/aeppl/blob/main/aeppl/mixture.py#L174) akin to the following snippet:

https://github.com/pymc-devs/pymc/blob/0fc84adf5f170138494815f9b8cb29e55923611f/pymc/distributions/mixture.py#L285-L288

We can introduce a `marginalize=False` argument to the `pm.Mixture` class to allow users to choose whether they want mixture indices or not and, internally, defer to either `MarginalMixtureRV` or `MixtureRV` in AePPL as necessary.

cc @ricardoV94

**Comment by lucianopaz** [comment]
I disagree with adding this to the mixture api. The users that want to sample the mixture indices can do this already. They only need to create an explicit `Categorical` rv that represents the index, and the mixture components as a regular rv, and use the index rv to select the component that goes downstream in the model.

**Comment by ricardoV94** [comment]
You could already do that but you cant condition on observations that way.

One advantage of bridging with Aeppl is that we don't need to write different models for deterministic operations that are observed vs unobserved.

Stacking and indexing is one such case that's already well supported by Aeppl and that PyMC users cant easily do otherwise.

**Comment by ricardoV94** [comment]
Besides the advantage of allowing for observations, I also don't see the problem with providing the same API that behind the scenes builds the categorical and stacking/ indexing operations for the users. 

This can lower the barrier for users that don't understand indexing and stacking operations, and we are already doing most of this work anyway for the random graph of the marginalized mixtures

**Comment by ricardoV94** [comment]
Also note that the logp created by PyMC for these non-marginalized Mixtures is very interesting and challenging for a user to create. It only evaluates the values at the needed mixtures, depending on the index variable. It works like a "lazy" switch.
